src/app/screens/edit-personal-info/index.tsx

```diff
@@ -0,0 +1,161 @@
+    defaultValues: {
+      firstName: profileData?.name,
+      lastName: profileData?.birthDate,
+      birthDate: profileData?.birthDate,
+      gender: profileData?.gender,
+    },
+    resolver: yupResolver(validationSchema),
+  });
+
+  const fetchProfileData = (): Promise<ProfileData> => {
+    return new Promise((resolve, reject) => {
+      setTimeout(async () => {
+        try {
+          resolve(profileDataFile);
+        } catch (error) {
+          reject(error);
+        }
+      }, 1000);
+    });
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const data = await fetchProfileData();
+        setProfileData(data);
+        reset({
+          firstName: data?.name || '',
+          lastName: data?.lastName || '',
+          birthDate: data?.birthDate || '',
+          gender: data?.gender || '',
+        });
+      } catch (error) {
+        console.error('Error fetching profile data:', error);
+      }
+    };
+
+    fetchData();
+  }, [profileData, reset]);
+
+  const onValidSubmit: SubmitHandler<FormData> = async data => {
+    try {
+      //TODO: dispatch thunks
+      console.log(data);
+    } catch (error: any) {
+      Alert.alert(t('screens:editPersonalData:error'), error.message);
+    }
+  };
+
+  return (
+    <SafeAreaView style={commonStyles.safeAreaViewStyle}>
+      <KeyboardAvoidingView
+        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+        style={commonStyles.keyboardAvoidingView}
+      >
+        <ScrollView>
+          <Box flex={1} paddingVertical="$10" paddingHorizontal="$12">
+            <Box alignItems="center" padding="$2" mb="$6">
+              <Text size="2xl" color="$textDark950">
+                {t('screens:editPersonalData.heading1')}
+              </Text>
+            </Box>
+            <Box>
+              <ControlledInput
+                controller={{
+                  control,
+                  name: 'firstName',
+                }}
+                formControlProps={{
+                  mb: '$4',
+                }}
+              />
+
+              <ControlledInput
+                controller={{
+                  control,
+                  name: 'lastName',
+                }}
+                formControlProps={{
+                  mb: '$4',
+                }}
+              />
+
+              <ControlledInput
+                controller={{
+                  control,
+                  name: 'birthDate',
+                }}
+                formControlProps={{
+                  mb: '$4',
+                }}
+              />
+              <ControlledSelectInput
+                controller={{
+                  control,
+                  name: 'gender',
+                }}
+                formControlProps={{
+                  mb: '$4',
+                }}
+                options={genderOptions}
+              />
+
+              <Button
+                onPress={() => navigation.goBack()}
```

Comment:
Wdyt if we do this here also?
```suggestion
                onPress={navigation.goBack}
```